core/src/main/scala/neotypes/implicits.scala

```diff
@@ -381,4 +383,46 @@ object implicits {
       new DeferredQueryBuilder(parts.toList)
     }
   }
+
+  /**
+    * Async
+    */
```

Comment:
I realized that it was strange that we had almost all implicits in this file, except for the ones related with `Asyc`.
Thus, I think that moving it here would be the best option, that way one only had to `import neotypes.implicits._` to use the library.

Reply:
I'm thinking that this file becomes overloaded with too many different types of implementations. As far as I remember, I saw some library where all implicits were broken down into multiple classes and defined as trait members, then all of them were mixed to `object implicits`

Reply:
Yeah, that is what cats and most typelevel projects do.
Sounds good, but will leave it for another **PR** this one already moves too many things.